src/librustc_mir/build/expr/stmt.rs

```diff
@@ -161,6 +162,41 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                 );
                 block.unit()
             }
+            ExprKind::PlaceTypeAscription { source, user_ty } => {
+                let place = unpack!(block = this.as_place(block, source));
+                this.cfg.push(
+                    block,
+                    Statement {
+                        source_info,
+                        kind: StatementKind::AscribeUserType(
+                            place,
+                            Variance::Invariant,
+                            user_ty,
+                        ),
+                    },
+                );
+                block.unit()
+            }
+            ExprKind::ValueTypeAscription { source, user_ty } => {
```

Comment:
Similarly this should go into `as_rvalue`, I think?

https://github.com/rust-lang/rust/blob/1002e404e1ae6b8b907c8655edd41380d0c561cb/src/librustc_mir/build/expr/as_rvalue.rs#L49-L54

Or maybe, actually, it too should go into `as_place`, and it should just evaluate its argument into a `temp` and yield the temp as a place (sort of like it does today).